### PR TITLE
Add some infrastructure for invalid CASE message testing.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Adds some basic infrastructure for this and a test that tests a
SigmaR1 with an empty payload.

#### Problem
Hard to test what happens with CASE if you send invalid messages right now.

#### Change overview
Add some test infrastructure for this.

#### Testing
Basic "send an invalid message" test added using the new infrastructure.

### CODE REVIEW

File: src/protocols/secure_channel/tests/TestCASESession.cpp [hunk 1]
```diff
@@ -432,6 +522,40 @@ void CASE_SecurePairingSerializeTest(nlTestSuite * inSuite, void * inContext)
...
+    private:
+        CHIP_ERROR AdjustMessage(const PayloadHeader & payloadHeader, System::PacketBufferHandle & msg) override
+        {
+            if (payloadHeader.HasMessageType(SecureChannel::MsgType::CASE_SigmaR1))
+            {
+                // Empty out the payload
+                msg->SetDataLength(0);
+            }
+            return CHIP_NO_ERROR;
+        }
```
Comment (tcarmelveilleux) [inline src/protocols/secure_channel/tests/TestCASESession.cpp:544]: This breaks the message, which helps test some more cases. But what about testing for different contents of the payload (e.g. not assuming that the sender of a message is doing things right, to test attack/fuzz situations)? How do we make test cases to validate sender data contents, and receiver parsing behavior?

Comment (bzbarsky-apple) [reply src/protocols/secure_channel/tests/TestCASESession.cpp:544]: With this setup, you basically need a different adjuster class (or perhaps an adjuster class parametrized on some data) per test you want to run, yes...

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @mspang @andy31415 @Damian-Nordic ?

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Comment by andy31415** [comment]
@bzbarsky-apple - conflicts on this one. Commenting before stale bot has a chance to close  it.

**Pull request closed without merging**